packages/backend/src/wallet/wallet.service.ts

```diff
@@ -85,6 +89,47 @@ export const createWallet = async (
   }
 }
 
+export const updateProfile = async (
+  req: Request,
+  res: Response<BaseResponse>
+) => {
+  try {
+    const { id } = req.user as User
+    let user = await User.query().findById(id)
+    if (!user) throw new Error(`user doesn't exist`)
+    const { firstName, lastName } = await zParse(profileSchema, req)
+
+    const result = await updateRapydProfile({
+      first_name: firstName,
+      last_name: lastName,
+      ewallet: user.rapydEWalletId
+    })
+
+    if (result.status.status !== 'SUCCESS')
+      return res.status(500).json({
+        message: `Unable to update profile : ${result.status.message}`,
+        success: false
+      })
+
+    user = await User.query().patchAndFetchById(id, {
```

Comment:
Should I just use update here? Or this function does an update and returns the updates user as well?

Reply:
👍
Reference: https://vincit.github.io/objection.js/api/query-builder/mutate-methods.html#patchandfetchbyid